### DISCUSSION TITLE
Welcome to geoai Discussions!

### DISCUSSION THREAD

**Discussion by giswqs** [opengeos/geoai#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by popogis24** [comment]
I’m Anderson Stolfi, an Python Developer. I founded [ilovegis.com.br](https://ilovegis.com.br/), a GIS report automation platform powered by LLMs, designed to streamline environmental and territorial analysis.

I’m really excited about the geoai project and I'm looking for opportunities to contribute :)

**Reply by popogis24** [reply]
btw, @giswqs do you have any discord channel for discussion about this project?

**Reply by giswqs** [reply]
Yes, there is a discord server. I don't use it very often. Most discussions happen on GitHub. 
https://github.com/opengeos

**Reply by popogis24** [reply]
Okay, thanks, Wu.
I just forked your GeoAI repo and am looking for features to contribute. Do you have anything specific you’d like me to focus on?

**Reply by giswqs** [reply]
Any contributations are welcome! You can submit PRs as you wish. We will go from there. If you have any questions, feel free to ask here.

**Reply by popogis24** [reply]
Hi Wu! I've been studying the codebase and learning a lot from it. I had a thought - and please let me know if this doesn't make sense - but maybe we could start modularizing some parts of the code? As a starting point, we could split the detection classes into separate files, like:

geoai/detectors/ - ship.py - car.py - solar_panel.py

This would just be a first step - there are probably other parts of the code we could organize better too. The code works great as it is now! I'm just thinking that as the library grows, having a more modular structure might make it easier to maintain and expand.

I'm still learning about best practices in library organization, so I'd really appreciate your thoughts on this. Would this be a helpful direction to explore?

**Reply by giswqs** [reply]
Yes, that makes sense. I have not gone that route yet because right now those detectors are just a thin wrapper of the `ImageDetector` class. Each detector only has a few lines of code, which does not really need to be put in separate modules yet.

**Reply by popogis24** [reply]
Oh, okay, right!

**Comment by konglingzhia** [comment]
Hello, may I ask how to obtain more datasets for "Building detection with aerial imagery" and "Building detection with aerial imagery and LiDAR data"?

**Comment by 1042422227** [comment]
Hola 👋 cómo estás

**Comment by Ragavi12-09** [comment]
<img width="459" height="469" alt="Screenshot from 2026-03-13 11-48-06" src="https://github.com/user-attachments/assets/b004da56-ff8c-41f3-afdb-4b4920d3848b" />

I am trying to install this plugin in my ubuntu machine QGIS version 3.40.14-Bratislava , during the dependencies installation, i am getting the above error, Please anyone guide me to what should I do next ?
